04-plots.R

```diff
@@ -0,0 +1,88 @@
+# Task 4
+
+library(tidyverse)
+library(ggplot2)
+install.packages("ggrepel")
+library(ggrepel)
+
+# Q1 Scatterplot of Sequence Lengths vs. Starting Integers --------------------
+
+# Filtering the top 10 results from collatz_df
+top10data <- collatz_df %>%
+  filter(start %in% top10longest)
+
+# Create complex plots from data in a data frame using ggplot
+ggplot(
+  data = collatz_df,
+  mapping = aes(
+    x = start,
+    y = length
+  )
+) +
+  # Create scatter plot and labelling
+  geom_point() +
+  labs(
+    title = "Starting Integers vs Sequence Lengths",
+    x = "Starting Integers",
+    y = "Sequence Lengths"
+  ) +
+  geom_point(
+    data = top10data, aes(x = start, y = length),
+    col = "red", size = 3, shape = 15
+  ) +
+  # To avoid overlaps by repelling text labels
+  geom_text_repel(
+    data = top10data,
+    aes(x = start, y = length, label = start),
+    vjust = -0.5,
+    hjust = 0.5,
+    size = 3
+  ) +
+  theme_classic()
+
+# Q2 Scatter plot of Highest Sequence Value vs. Starting Integers -------------
+
+# Identifying the top 10 starting integers (highest value reached)
+top10highvalue <- collatz_df %>%
+  # rearranging the data frame or starting integers in descending manners
+  arrange(desc(max_val)) %>%
+  # selecting top 10 starting integers from above code
+  top_n(10)
+
+# Create the scatterplot for highest value vs. starting integers
```

Comment:
The scatterplot is kind of clustered and difficult to visualize? 
Maybe changing the size and colour of the plot helps?